### PR TITLE
[build]: add the support of Apt-Cacher-NG proxy configuration to the …

### PR DESCRIPTION
**Pull request by wangshengjun**
[build]: add the support of Apt-Cacher-NG proxy configuration to the …
Signed-off-by: wangshengjun <wangshengjun@asterfusion.com>

What I did
Add the support of Apt-Cacher-NG proxy configuration to the slave docker build script.

Why I did it
The build process of basic slave docker always takes a long time to download the apt/debian package from the internet. We can introduce the apt cacher server to cut down the download time extremely.
So it makes sense  to add the support of Apt-Cacher-NG proxy configuration.

How I verified it
I set up the apt cacher server in my local area network(IP :10.250.0.240, Port : 3142)
Build cmd as follows:
with apt_cacher_ng_proxy args
apt_cacher_ng_proxy=http://10.250.0.240:3142 make sonic-slave-build
without apt_cacher_ng_proxy args
make sonic-slave-build
Both works well.

### CODE REVIEW

**Comment by lguohan** [comment]
the proxy only works for the slave docker build, what about image build and docker build, can you add cache ng proxy for those as well?

**Comment by lguohan** [comment]
since this cache ng proxy only does apt cache, does it conflict with http_proxy option? Since http_proxy can be also used for pip package download.

**Comment by wangshengjun** [comment]
> since this cache ng proxy only does apt cache, does it conflict with http_proxy option? Since http_proxy can be also used for pip package download.

the cache ng proxy only does apt cache, it works fine with http_proxy option.

**Comment by wangshengjun** [comment]
> the proxy only works for the slave docker build, what about image build and docker build, can you add cache ng proxy for those as well?

Yes, the apt-cacher-ng proxy can add into the 'docker-base-strecth', so other dockers can inherit.

File: Makefile.work [hunk 1]
```diff
@@ -173,6 +175,7 @@ SONIC_BUILD_INSTRUCTION :=  make \
                            KERNEL_PROCURE_METHOD=$(KERNEL_PROCURE_METHOD) \
                            HTTP_PROXY=$(http_proxy) \
                            HTTPS_PROXY=$(https_proxy) \
+		           APT_CACHER_NG_PROXY=$(apt_cacher_ng_proxy) \
```
Comment (qiluo-msft) [inline Makefile.work]: >		            [](start = 0, length = 13)

The indentation is not same as above lines

File: dockers/docker-base/Dockerfile.j2 [hunk 2]
```diff
@@ -23,6 +23,10 @@ RUN rm -rf               \
     /var/cache/man/*     \
     /usr/share/locale/*
 
+{% if APT_CACHER_NG_PROXY.strip() -%}
+        RUN echo 'Acquire::http::Proxy "{{APT_CACHER_NG_PROXY}}";' >> /etc/apt/apt.conf.d/02proxy
```
Comment (qiluo-msft) [inline dockers/docker-base/Dockerfile.j2]: >/etc/apt/apt.conf.d/02proxy [](start = 70, length = 27)

Does it impact runtime proxy? If yes, production switches may have no access to build environment at all.

Comment (qiluo-msft) [inline dockers/docker-base/Dockerfile.j2]: >         [](start = 0, length = 8)

Suggest remove the leading spaces.

File: dockers/docker-base/Dockerfile.j2 [hunk 3]
```diff
@@ -23,6 +23,10 @@ RUN rm -rf               \
     /var/cache/man/*     \
     /usr/share/locale/*
 
+{% if APT_CACHER_NG_PROXY.strip() -%}
```
Comment (qiluo-msft) [inline dockers/docker-base/Dockerfile.j2]: >APT_CACHER_NG_PROXY [](start = 6, length = 19)

Suggest rename to APT_PROXY, Apt-Cacher-NG is a specific apt proxy implementation.

**Comment by lguohan** [comment]
can you do the rebase and resolve the conflict? it is difficult to figure out what is the change now.

**Pull request closed without merging**